Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs

```diff
@@ -1,148 +1,179 @@
             _contents.Capabilities = SolutionCaps.PourIn
                                      | SolutionCaps.PourOut
                                      | SolutionCaps.Injectable;
-            _drinking = false;
-            Owner.TryGetComponent(out AppearanceComponent appearance);
-            _appearanceComponent = appearance;
-            _appearanceComponent?.SetData(SharedFoodComponent.FoodVisuals.MaxUses, MaxVolume.Float());
-            _updateAppearance();
+            _opened = _defaultToOpened;
+            _originalName = Owner.Name;
+            if (_opened)
+            {
+                UpdateName();
+            }
+
+            UpdateAppearance();
         }
 
-        private void _updateAppearance()
+        void ISolutionChange.SolutionChanged(SolutionChangeEventArgs eventArgs)
         {
-            _appearanceComponent?.SetData(SharedFoodComponent.FoodVisuals.Visual, _contents.CurrentVolume.Float());
+            UpdateName();
+            UpdateAppearance();
         }
 
-        bool IUse.UseEntity(UseEntityEventArgs eventArgs)
+        private void UpdateName()
         {
-            UseDrink(eventArgs.User);
+            if (Opened)
+            {
+                if (Empty)
+                {
+                    Owner.Name = $"{_originalName} {Loc.GetString("(empty)")}";
+                }
+                else
+                {
+                    Owner.Name = $"{_originalName} {Loc.GetString("(opened)")}";
+                }
+            }
 
-            return true;
         }
 
+        private void UpdateAppearance()
+        {
+            _appearanceComponent?.SetData(SharedFoodComponent.FoodVisuals.Visual, _contents.CurrentVolume.Float());
+        }
+        bool IUse.UseEntity(UseEntityEventArgs args)
+        {
+            if (!_opened)
+            {
+                //Do the opening stuff like playing the sounds.
+                var soundCollection = _prototypeManager.Index<SoundCollectionPrototype>(_soundCollection);
+                var file = _random.Pick(soundCollection.PickFiles);
+                _audioSystem = _entitySystem.GetEntitySystem<AudioSystem>();
+                _audioSystem.Play(file, Owner, AudioParams.Default);
+                _opened = true;
+                UpdateName();
+                return false;
+            }
+            return TryUseDrink(args.User);
+        }
+
+        //Force feeding a drink to someone.
         void IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
         {
-            if (!InteractionChecks.InRangeUnobstructed(eventArgs)) return;
+            TryUseDrink(eventArgs.Target);
+        }
 
-            UseDrink(eventArgs.Target);
+        public void Examine(FormattedMessage message)
+        {
+            var color = Opened ? "yellow" : "white";
+            var openedText = Opened ? Loc.GetString("opened") : Loc.GetString("closed");
+            message.AddMarkup(Loc.GetString($"It is [color={color}]{openedText}[/color].") );
```

Comment:
This should be `Loc.GetString("It is [color={0}]{1}[/color].", color, openedText)` instead.

Reply:
Thanks thats a lot better.